### ISSUE TITLE
The nuget package: https://www.nuget.org/api/v2/package/System.Buffers/4.5.1 has mismatched assembly versions.

### ISSUE THREAD

**Issue by KevinRansom** [dotnet/runtime#47949]
The nuget package: https://www.nuget.org/api/v2/package/System.Buffers/4.5.1 has mismatched assembly versions.

Which requires binding redirects on the desktop to successfully bind.

Download the nuget package : https://www.nuget.org/api/v2/package/System.Buffers/4.5.1

Extract the files:
use ildasm or some other mechanism to examine the assembly version numbers.
````
ref\net45\System.Buffers.dll                      Version:   4.0.3.0
ref\netstandard1.1\System.Buffers.dll       Version:   4.0.2.0
ref\netstandard2.0\System.Buffers.dll       Version:   4.0.2.0

lib\net461\System.Buffers.dll                    Version:   4.0.3.0
````
This is  an issue because if I build a library for netstandard2.0, and reference it from an app built for net472 the app needs a binding redirect to correctly execute.

**Comment by huoyaoyuan** [comment]
https://github.com/dotnet/runtime/issues/45921

**Comment by danmosemsft** [comment]
@Anipik

**Comment by saucecontrol** [comment]
Not a direct fix, but if you're targeting .NET Standard and intend compatibility with .NET Framework, your users will have a better experience in general if you multi-target your library for both `net461` and `netstandard2.0`.  That will fix your issue with System.Buffers as a side effect, but there are other benefits as well.

Rick Strahl has a good writeup on that [here](https://weblog.west-wind.com/posts/2019/Feb/19/Using-NET-Standard-with-Full-Framework-NET), and the same recommendation is also given in Microsoft's guidance for library developers [here](https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/cross-platform-targeting#multi-targeting).

**Comment by KevinRansom** [comment]
@saucecontrol ,
If I build and reference ref/netstandard2.0 and the assembly verision is  4.0.2.0 and The corresponding lib/netstandard2.0 has an assembly version of netstandard 4.0.3.0 then on the desktop a binding redirect is necessary.

For developers building software that is a plugin, it is very hard to get an app to add the necessary binding redirect.  I would suggest that having different version numbers for ref and lib in the same nuget package is just broken.

**Comment by saucecontrol** [comment]
Correct, if the versions don't match, the binding redirect is necessary.  However, if you multi-target, the reference in the `net461` version of your library in the package would be 4.0.3 while the `netstandard2.0` copy of your library would reference 4.0.2.  I've been through this in my own libraries, and multi-targeting, along with careful selection of dependency versions has solved the issue for my users.

> having different version numbers for ref and lib in the same nuget package is just broken.

I couldn't agree more, and I said as much over on https://github.com/dotnet/runtime/issues/26370.  You might find that issue informative, if only so you can get a picture of how/why the current broken state came to be ;)
